### DISCUSSION TITLE
Attribute Error: Spacy version 2.3

### DISCUSSION THREAD

**Discussion by Ajit-Solanki** [explosion/spaCy#11799] [Help: Coding & Implementations]
Hello, 
I am using medspacy with 0.2.0 and spacy 2.3, using for some pattern matching function using regex. Made new component added in pipeline.
Throwing an error on line
""Text = doc.text""
"English" object has no attribute "text"
Same also, "English" object has no attribute "ents".
If anyone could help out, i can also post full code .

Thanks and regards

**Comment by polm** [comment]
Sorry you're having trouble with this. Can you give a short sample of code and the full stack trace you're getting?

Based on your error, note it looks like you're doing something like `nlp.text`, which won't work - the `nlp` object you get by calling `spacy.load` is of type Language/English, but a normal Doc object is not. 

Also note that since v0.2.0 medspaCy supports spaCy v3, so I would recommend you upgrade spaCy.

**Comment by Ajit-Solanki** [comment]
Thanks @polm for reverting. Perhaps, i found the solution for that, had an issue while creating custom label. Now i have one more query.

Here's to the sample code example.
I have an object entity "ent" which contains some name "John Bryan is a " now i am converting it to text by ent.text just to eliminate extra content from ent like i want John Bryan only. Is it possible to convert this "John Bryan"  text of "str" type back to ent type object. If so how can I . Any solution would be appreciated. 
Thanks & Regards,
Aj

**Reply by polm** [reply]
Entities you get a from a Doc object are of type Span, there is no "ent type". You can't turn a string back to a Span arbitrarily, a Span has to be associated with some Doc. 

It might help if you could explain why you want to convert things back in more detail.

**Reply by Ajit-Solanki** [reply]
Yes , reason is i just want to convert text back into span to find the geometry coordinates of extracted name.

**Reply by polm** [reply]
OK, in that case you need to keep the original span around. With just the text there's no way to tell the difference between two occurrences in different places, for example.

**Reply by Ajit-Solanki** [reply]
Okay. Let me elaborate in details:-
1. I am using regex and creating one custom component to match and find people name from doc. 
2. From ent in doc.ents what am getting is like " person name : John Bryan" 
3. Now i want to remove extra / unnecessary things and keep only "John Bryan" , so converted ent to text by ent.text and removed "person name:" from the text , left with only "John Bryan" 
4. The found name will be used to re.findall from same document. And return name "John Bryan" from whole page , this process repeats for every pages in document.
5. For name we found i want to get geometry coordinates. So that I can create bounding box on the same.

This is the complete flow , after finding name return values are in text not in ent span , concern is how can I find the coordinates...of text ...

I hope you'll get the flow.

**Reply by polm** [reply]
OK, thank you for explaining the situation. It sounds like the issue is that in the final extraction step you use `re.findall`, which just gives you strings, and does not tell you where in the document they are. I would suggest that you instead use a PhraseMatcher, which will match exact phrases and give you a Span. 

I am not sure how you would map Spans to position in the document - spaCy won't help you with that - but it would be easier than using just a string. I guess you could treat each paragraph as a Doc and associate some position information with it?

To step back a little:

> From ent in doc.ents what am getting is like " person name : John Bryan"

That's kind of weird. This entity is assigned by your custom component, using regex matches? Maybe it would be better to modify your component so it only marks "John Bryan" as an entity? You might also look into using an NER component.
